test/test_torch.py

```diff
@@ -4786,6 +4786,33 @@ def test_equal(self):
         self.assertTrue(torch.equal(s1, s3))
         self.assertFalse(torch.equal(s1, s4))
 
+    def test_comparator_integral_tensor_floating_scalar(self):
+        t = torch.LongTensor([-2, -1, 0, 1, 2])
+        self.assertEqual(t.lt(0.5), [1, 1, 1, 0, 0])
+        self.assertEqual(t.lt(-0.5), [1, 1, 0, 0, 0])
+        self.assertEqual(t.gt(0.5), [0, 0, 0, 1, 1])
+        self.assertEqual(t.gt(-0.5), [0, 0, 1, 1, 1])
+        self.assertEqual(t.ge(0.5), [0, 0, 0, 1, 1])
+        self.assertEqual(t.ge(-0.5), [0, 0, 1, 1, 1])
+        self.assertEqual(t.le(0.5), [1, 1, 1, 0, 0])
+        self.assertEqual(t.le(-0.5), [1, 1, 0, 0, 0])
+        self.assertEqual(t.eq(0.5), [0, 0, 0, 0, 0])
+        self.assertEqual(t.ne(0.5), [1, 1, 1, 1, 1])
+
+    @unittest.skipIf(not torch.cuda.is_available(), 'no CUDA')
+    def test_comparator_integral_tensor_floating_scalar_cuda(self):
+        t = torch.LongTensor([-2, -1, 0, 1, 2]).cuda()
+        self.assertEqual(t.lt(0.5), [1, 1, 1, 0, 0])
+        self.assertEqual(t.lt(-0.5), [1, 1, 0, 0, 0])
+        self.assertEqual(t.gt(0.5), [0, 0, 0, 1, 1])
+        self.assertEqual(t.gt(-0.5), [0, 0, 1, 1, 1])
+        self.assertEqual(t.ge(0.5), [0, 0, 0, 1, 1])
+        self.assertEqual(t.ge(-0.5), [0, 0, 1, 1, 1])
+        self.assertEqual(t.le(0.5), [1, 1, 1, 0, 0])
+        self.assertEqual(t.le(-0.5), [1, 1, 0, 0, 0])
+        self.assertEqual(t.eq(0.5), [0, 0, 0, 0, 0])
+        self.assertEqual(t.ne(0.5), [1, 1, 1, 1, 1])
```

Comment:
These two tests are identical. A better way would be to write a helper that takes a `dtype`, and change these methods to:
```python
def test_comparator_integral_tensor_floating_scalar(self):
    self._test_comparator_integral_tensor(torch.long) # or torch.cuda.long in CUDA test
```

Reply:
Also, CUDA tests should be in `test_cuda.py`, not in `test_torch.py`. There are some other examples (grep for `TestTorch` in `test_cuda.py`)